src/TextField/TextField.js

```diff
@@ -105,7 +105,7 @@ const getStyles = (props, context, state) => {
  * @returns True if the string provided is valid, false otherwise.
  */
 function isValid(value) {
-  return value !== '' && value !== undefined && value !== null;
+  return value !== '' && value !== undefined && value !== null && !(Array.isArray(value) && value.length === 0);
```

Comment:
I 100% agree with the motivation. My point is about moving the piece of logic into the DropDownMenu component.